ISPN-12763 Load configuration URIs from classpath

tristantarrant opened the pull request:
https://issues.redhat.com/browse/ISPN-12763

danberindei commented on commons/all/src/main/java/org/infinispan/commons/jdkspecific/ClasspathURLStreamHandler.java:
```diff
@@ -0,0 +1,47 @@
+               String path = u.getPath();
+               URL resource = ClassLoader.getSystemClassLoader().getResource(path);
+               if (resource == null) {
+                  resource = Thread.currentThread().getContextClassLoader().getResource(path);
```
The context class loader may be `null`. 

IMO it would be better to try the context class loader first (if non-null) and fall back to the system class loader.

danberindei commented on commons/jdk11/src/main/java/org/infinispan/commons/jdkspecific/ClasspathURLStreamHandler.java:
```diff
@@ -0,0 +1,54 @@
+ * @since 12.1
+ **/
+@MetaInfServices
+public class ClasspathURLStreamHandler extends URLStreamHandlerProvider {
```
How does this implementation differ from the pre-java 11 implementation?

It would be nice to have a description in the javadoc.

tristantarrant commented:
@danberindei I've reworked it as follows:
* the `all` version is now in the `org.infinispan.commons.util` package and is public API
* the `jdk11` version stays in the `jdkspecific` package and is only there to provide a `URLStreamHandlerProvider` via service loader
I've also addressed the classloader order and null checks

danberindei commented:
Merged, thanks Tristan!
